packages/keystatic/src/component-blocks/cloud-image-preview.tsx

```diff
@@ -10,65 +10,189 @@ import { Icon } from '@keystar/ui/icon';
-      props.onChange({
-        src: match[2],
-        alt: match[1],
-        height: '',
-        width: '',
-      });
+      setPastedData(
+        cleanImageData({
+          src: match[2],
+          alt: match[1],
+        })
+      );
+      return;
+    }
+    setPastedData(cleanImageData({ src: text }));
+  };
+
+  useEffect(() => {
+    if (!pastedData || !pastedData.src) {
+      setStatus('');
+      return;
+    }
+    if (!isValidURL(pastedData.src)) {
+      setStatus('The pasted data is not a valid Image URL');
       return;
     }
+    const img = new Image();
+    setStatus('Checking pasted data...');
+    img.onload = () => {
+      onChange({
+        ...pastedData,
+        width: img.width.toString(),
+        height: img.height.toString(),
+      });
+    };
+    img.onerror = () => {
+      setStatus('The pasted data is not a valid Image URL');
+      console.log('Invalid Image URL pasted:', pastedData);
+    };
+    img.src = pastedData.src;
+    return () => {
+      img.onload = null;
+    };
+  }, [pastedData, onChange]);
+
+  return (
+    <NotEditable>
+      <VStack gap="regular">
+        <div style={{ position: 'relative' }}>
+          <input
+            type="text"
+            value=""
+            onFocus={() => setIsFocused(true)}
+            onBlur={() => setIsFocused(false)}
+            onClick={() => {
+              document.execCommand('paste');
+            }}
+            onPaste={onPaste}
+            style={{
+              position: 'absolute',
+              cursor: 'default',
+              fontSize: '16px',
+              border: 'none',
+              outline: isFocused ? '2px solid #30a46c' : 'none',
+              background: 'transparent',
+              padding: 0,
+              paddingLeft: '2%',
+              margin: 0,
+              opacity: 0,
+              color: 'transparent',
+              borderRadius: '6px',
+              width: '98%',
+              height: '100%',
+            }}
+          />
+          {isFocused ? (
+            <Flex
+              backgroundColor="positive"
+              border="positive"
+              color="positive"
+              justifyContent="center"
+              borderRadius="regular"
+              padding="regular"
+            >
+              {isMobile ? 'Tap' : `Press ${metaSymbol} + V`} to paste an Image
+              URL
+            </Flex>
+          ) : (
+            <Flex
+              border="neutral"
+              color="neutral"
+              justifyContent="center"
+              borderRadius="regular"
+              padding="regular"
+            >
+              {isMobile ? 'Tap' : 'Click'} here to paste an Image URL
+            </Flex>
+          )}
+        </div>
+        {config.storage.kind === 'cloud' && (
+          <div>
+            <ActionButton
+              href={getProjectImagesURL(config.storage.project)}
+              target="_blank"
+              rel="noreferrer noopener"
```

Comment:
Nit: `noopener` is [implied](https://html.spec.whatwg.org/multipage/links.html#link-type-noreferrer) by `noreferrer`.